native/cocos/ui/edit-box/EditBox-ios.mm

```diff
@@ -363,76 +265,315 @@ - (BOOL)textFieldShouldReturn:(UITextField *)textField {
+    return ret;
 }
 
-- (void)textViewDidChange:(UITextView *)textView {
-    if (textView.markedTextRange != nil)
-        return;
+//TODO: show inputbox with width modified.
+- (void) show: (const cc::EditBox::ShowInfo*)showInfo {
+    g_maxLength = showInfo->maxLength;
+    g_isMultiline = showInfo->isMultiline;
+    
+    if (g_isMultiline) {
+        curView = [self createTextView:showInfo];
+    } else {
+        curView = [self createTextField:showInfo];
+    }
+    [[curView inputOnView] setHidden:INPUTBOX_HIDDEN];
+    UIView *view = UIApplication.sharedApplication.delegate.window.rootViewController.view;
+    
+    [view addSubview:[curView inputOnView]];
+    [[curView inputOnView] becomeFirstResponder];
+    [[curView inputOnToolbar] becomeFirstResponder];
+}
+// Change the focus point to the TextField or TextView on the toolbar.
+- (void) hide {
+    [[curView inputOnView] becomeFirstResponder];
+    [[curView inputOnView] removeFromSuperview];
+    [[curView inputOnToolbar] resignFirstResponder];
+    [[curView inputOnView] resignFirstResponder];
+}
 
-    // check length limit after text changed, a little rude
-    if (textView.text.length > g_maxLength)
-        textView.text = [textView.text substringToIndex:g_maxLength];
+- (InputBoxPair*) getCurrentViewInUse {
+    return curView;
+}
+- (NSString*) getCurrentText {
+    if (g_isMultiline)
+        return [(UITextView*)[curView inputOnToolbar] text];
+    return [(UITextField*)[curView inputOnToolbar] text];
+}
 
-    callJSFunc("input", [textView.text UTF8String]);
+
+@end
+@implementation ButtonHandler
+- (IBAction)buttonTapped:(UIButton *)button {
+    const ccstd::string text([[[EditboxManager sharedInstance]getCurrentText] UTF8String]);
+    callJSFunc("confirm", text);
+    cc::EditBox::complete();
 }
 @end
-
 /*************************************************************************
  Implementation of EditBox.
  ************************************************************************/
 
 // MARK: EditBox
 
-namespace cc {
-
+namespace cc{
 bool EditBox::_isShown = false;
-
 void EditBox::show(const cc::EditBox::ShowInfo &showInfo) {
-    // Should initialize them at first.
-    g_maxLength = showInfo.maxLength;
-    g_isMultiline = showInfo.isMultiline;
-    g_confirmHold = showInfo.confirmHold;
-
-    addKeyboardEventLisnters();
-    addTextInput(showInfo);
-
-    _isShown = true;
+    [[EditboxManager sharedInstance] show:&showInfo];
+    EditBox::_isShown = true;
 }
 
 void EditBox::hide() {
-    removeKeyboardEventLisnters();
-
-    UIView *view = getCurrentView();
-    if (view) {
-        [view removeFromSuperview];
-        [view resignFirstResponder];
-    }
-
-    _isShown = false;
+    [[EditboxManager sharedInstance] hide];
+    EditBox::_isShown = true;
 }
 
 bool EditBox::complete() {
-    if (!_isShown)
-        return false;
-
-    NSString *text = getCurrentText();
+    if(!EditBox::_isShown)
+        return;
```

Comment:
return void ?  Should return false instead.

Reply:
well, it's actually better to make function void because the behavior does not need to check the stats.

But right there I will return true here.